gneiss/balances.py

```diff
@@ -52,7 +54,7 @@ def _balance_basis(tree_node):
         basis[i, :] = np.array([0]*k[i] + [a[i]]*r[i] + [b[i]]*s[i] + [0]*t[i])
     # Make sure that the basis is in level order
     basis = basis[:, ::-1]
-    nds = list(nds)
+    nds = [n.name for n in nds]
```

Comment:
Note that we are only keep track of names instead of entire TreeNode objects.  This is to ensure consistency between the methods and immutability.